src/display/annotation_layer.js

```diff
@@ -1832,157 +1851,229 @@ class ChoiceWidgetAnnotationElement extends WidgetAnnotationElement {
 }
 
 class PopupAnnotationElement extends AnnotationElement {
-  // Do not render popup annotations for parent elements with these types as
-  // they create the popups themselves (because of custom trigger divs).
-  static IGNORE_TYPES = new Set([
-    "Line",
-    "Square",
-    "Circle",
-    "PolyLine",
-    "Polygon",
-    "Ink",
-  ]);
-
   constructor(parameters) {
-    const { data } = parameters;
-    const isRenderable =
-      !PopupAnnotationElement.IGNORE_TYPES.has(data.parentType) &&
-      !!(data.titleObj?.str || data.contentsObj?.str || data.richText?.str);
-    super(parameters, { isRenderable });
+    const { data, elements } = parameters;
+    const isRenderable = !!(
+      data.titleObj?.str ||
+      data.contentsObj?.str ||
+      data.richText?.str
+    );
+    super(parameters, { isRenderable, useRect: false });
+    this.elements = elements;
   }
 
   render() {
     this.container.classList.add("popupAnnotation");
 
-    const parentElements = this.layer.querySelectorAll(
-      `[data-annotation-id="${this.data.parentId}"]`
-    );
-    if (parentElements.length === 0) {
-      return this.container;
+    let open = this.data.open;
+    if (typeof PDFJSDev !== "undefined" && PDFJSDev.test("TESTING")) {
+      open = true;
```

Comment:
Nit: Perhaps make this a const instead?
```suggestion
    const open =
      typeof PDFJSDev !== "undefined" && PDFJSDev.test("TESTING")
        ? true
        : this.data.open;
```